### PR TITLE
Stateless Clients: Repricing SLOAD and SSTORE to pay for block proofs

### PR DESCRIPTION
**Pull request by AlexeyAkhunov**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-X.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your Github username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-2035.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+type: Standards Track
+category: Core
+created: 2019-05-16
+---
+
+<!--You can leave these HTML comments in your merged EIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new EIPs. Note that an EIP number will be assigned by an editor. When opening a pull request to submit your EIP, please use an abbreviated title in the filename, `eip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the EIP.-->
+The gas cost of EVM opcodes `SLOAD` and `SSTORE` increases in order to accomodate extra bandwith required to propagate block proof together with the block
```
Comment (axic) [inline EIPS/eip-2035.md]: typo: accomodate

File: EIPS/eip-2035.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+EIP-2028 (link will be provided when EIP is merged) describes the model that can be used for deciding the gas cost of data transmission. It is relevant
+because in the stateless client regime `SSTORE` and `SLOAD` operations add more data to be transmitted (as well as computation to verify the proofs).
+
+The main alternate design is the rent proportional to the size of the contract storage, which unfortunately introduces a serious griefing
+vulnerability problem, and so far the solution seems to be in redesigning and rewriting smart contracts in a way, which makes them not vulnerable.
+However, this approach is likely to be very expensive on the non-technical (ecosystem) level.
+
+## Backwards Compatibility
+<!--All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. EIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+This change is not backwards compatible and requires hard fork to be activated.
```
Comment (axic) [inline EIPS/eip-2035.md]: This section supposed to explain what adverse effects this change can have on already deployed contracts.

File: EIPS/eip-2035.md (hunk 1 above)
Comment (AlexeyAkhunov) [reply EIPS/eip-2035.md]: Thanks, done

File: EIPS/eip-2035.md (hunk 2 above)
Comment (AlexeyAkhunov) [reply EIPS/eip-2035.md]: I have added some initial description

Comment (axic) [reply EIPS/eip-2035.md]: Does this sentence needs to be kept in? It doesn't seem to be useful.

File: EIPS/eip-2035.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+Not very formal at the moment, but will be formalised with more research and prototyping. Gas of operations `SLOAD` and `SSTORE` increases by `X` gas
+for those operations that access storage slots that were not previously accessed during the same transaction.
+
+Future variant (will be possible after the implentation of the *Gross contract size acccounting*) is researched, where the increase is varied
+depending on the size of the contract storage, i.e. `SLOAD` and `SSTORE` for smaller contracts will be cheaper.
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+[EIP-1884](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1884.md) seeks to increase the gas cost of `SLOAD` but using a different justification
```
Comment (axic) [inline EIPS/eip-2035.md]: Please use the https://eips.ethereum.org/EIPS/eip-1884 url.

File: EIPS/eip-2035.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+(latency of the execution as described in the Motivation). This EIP is likely to increase the cost of `SLOAD` by a larger amount, therefore partially
+(because EIP-1884 also proposed other increases) supersedes EIP-1884.
+
+EIP-2028 (link will be provided when EIP is merged) describes the model that can be used for deciding the gas cost of data transmission. It is relevant
```
Comment (axic) [inline EIPS/eip-2035.md]: Can you make EIP-2028 links to the PR then?

File: EIPS/eip-2035.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+Instead, bandwidth consumption goes up. There is emprical analysis (unpublished, but will be) suggesting that 1 uncached `SSTORE` or `SLOAD` adds
+at most 1 kB to the block proofs. At the current cost of data transmission (68 gas per byte), this translates to the increase of gas cost of both
+operations by 69k gas. However, in light of proposal in EIP-2028 (link will be provided when EIP is merged), the increase can be made much smaller.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+Not very formal at the moment, but will be formalised with more research and prototyping. Gas of operations `SLOAD` and `SSTORE` increases by `X` gas
+for those operations that access storage slots that were not previously accessed during the same transaction.
```
Comment (axic) [inline EIPS/eip-2035.md]: Can you at least give a list of affected opcodes?

File: EIPS/eip-2035.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+The main alternate design is the rent proportional to the size of the contract storage, which unfortunately introduces a serious griefing
+vulnerability problem, and so far the solution seems to be in redesigning and rewriting smart contracts in a way, which makes them not vulnerable.
+However, this approach is likely to be very expensive on the non-technical (ecosystem) level.
+
+## Backwards Compatibility
+<!--All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. EIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+This change is not backwards compatible and requires hard fork to be activated.
+There might also be an adverse effect of this change on the already deployed contract. It is expected that after this EIP and
+EIP-2026 (rent prepayment for accounts), the recommendation will be made to raise the gas limit. This can somewhat dampen the
```
Comment (axic) [inline EIPS/eip-2035.md]: Please make this a link also.

File: EIPS/eip-2035.md (hunk 2 above)
Comment (AlexeyAkhunov) [reply EIPS/eip-2035.md]: There are some changes that do not require hard fork, but still are requirements for other changes, so I say the sentence is still useful

File: EIPS/eip-2035.md (hunk 5 above)
Comment (AlexeyAkhunov) [reply EIPS/eip-2035.md]: The affected opcodes are SLOAD and SSTORE

Comment (axic) [reply EIPS/eip-2035.md]: I'm not sure I understand this sentence then. 

> Gas of operations `SLOAD` and `SSTORE` increases by `X` gas for those operations that access storage slots that were not previously accessed during the same transaction.

Is something missing here?

**Comment by axic** [comment]
@AlexeyAkhunov fixed typos and formatting, but one question remains

File: EIPS/eip-2035.md (hunk 5 above)
Comment (AlexeyAkhunov) [reply EIPS/eip-2035.md]: You are right, it was not very clear, I changed it now, hope it is better. Thank you!

**Pull request merged**